### DISCUSSION TITLE
Reenable watch mode

### DISCUSSION THREAD

**Discussion by espoal** [rome/tools#4421] [Suggestions]
It seems in the [past](https://github.com/rome/tools/issues/1664) rome had a watch mode. It would be nice to reenable it:

- It would allow to add basic support for unsupported IDE, like webstorm. The server could watch, and automatically fix files, without any further IDE integration needed
- It would allow me to containerize rome for development purposes

**Comment by ematipico** [comment]
I can't recall a watch mode, but it would be interesting to understand more.

What kinda of behaviour would you expect to have? 

If you're looking for IDE integrations, Rome has a command called `lsp-proxy`.

**Reply by espoal** [reply]
Right now I can only run batch mode: I run the command `rome check svcs tests libs --apply-unsafe --use-server` and it fixes the files.
Instead it would be preferable to have a watch mode, ie adding a `--watch` flag so that anytime a file changes the daemon will format that file. We already have a long running process, the server, so it shouldn't be too hard to register `inotify` watchers to detect changes and hence retrigger the linting only where needed.
It seems in the [past](https://github.com/rome/tools/issues/1664) it was available.

**Reply by espoal** [reply]
I looked up LSP, looks really cool, but unfortunately WebStorm doesn't seem to [support](https://intellij-support.jetbrains.com/hc/en-us/community/posts/360001627960-IntelliJ-support-for-Language-Server-Protocol-as-a-client-) it ;(
